freyja/read_analysis_tools.py

```diff
@@ -0,0 +1,268 @@
+        insertions = [s.strip() for s in mutations[1].split(',')]
+        deletions = [s.strip() for s in mutations[2].split(',')]
+    try:
+        # parse tuples from indels
+        if len(insertions[0]) > 0:
+            insertions = [
+                (int(s.split(':')[0][1:]), s.split(':')[1][1:-2].strip('\''))
+                for s in insertions
+            ]
+
+        if len(deletions[0]) > 0:
+            deletions = [
+                (int(s.split(':')[0][1:]), int(s.split(':')[1][:-1]))
+                for s in deletions
+            ]
+
+        # get loci for all mutations
+        snp_sites = [int(m[1:len(m)-1])-1 for m in snps if m]
+        indel_sites = [s[0] for s in insertions if s] +\
+            [s[0] for s in deletions if s]
+    except ValueError:
+        print('extract: Error parsing', query_mutations)
+        print('extract: See README for formatting requirements.')
+        return -1
+
+    snp_dict = {int(mut[1:len(mut)-1])-1: mut[-1] for mut in snps if mut}
+
+    samfile = pysam.AlignmentFile(input_bam, 'rb')
+
+    reads_considered = []
+
+    all_sites = snp_sites + indel_sites
+    all_sites.sort()
+
+    for site in all_sites:
+        itr = samfile.fetch("NC_045512.2", site, site+1)
+
+        for x in itr:
+            ref_pos = set(x.get_reference_positions())
+            start = x.reference_start
+            sites_in = list(ref_pos & set(snp_sites))
+
+            seq = x.query_alignment_sequence
+            cigar = re.findall(r'(\d+)([A-Z]{1})', x.cigarstring)
+
+            # note: inserts add to read length, but not alignment coordinate
+            if 'I' in x.cigarstring:
+                i = 0
+                del_spacing = 0
+                ins_spacing = 0
+                ranges = []
+                insert_found = False
+                for m in cigar:
+                    if m[1] == 'I':
+                        if (start+i+del_spacing-ins_spacing,
+                                seq[i:i+int(m[0])]) in insertions:
+                            reads_considered.append(x.query_name)
+                            insert_found = True
+                            break
+                        i += int(m[0])
+                        ins_spacing += int(m[0])
+                    elif m[1] == 'D':
+                        del_spacing += int(m[0])
+                        continue
+                    elif m[1] == 'M':
+                        ranges.append([i, i+int(m[0])])
+                        i += int(m[0])
+
+                seq = ''.join([seq[r[0]:r[1]] for r in ranges])
+
+                if insert_found:
+                    continue
+
+            if 'D' in x.cigarstring:
+                c_dict = dict(zip(x.get_reference_positions(), seq))
+                i = x.reference_start
+
+                for m in cigar:
+                    if m[1] == 'M':
+                        i += int(m[0])
+                    elif m[1] == 'D':
+                        for k in range(i, i+int(m[0])):
```

Comment:
I don't think we need to add the deletion to c_dict here. Probably a holdover from the old script?